packages/manager/modules/pci/src/projects/project/instances/instance/shelve/shelve.controller.js

```diff
@@ -0,0 +1,76 @@
+import get from 'lodash/get';
+
+export default class PciInstanceShelveController {
+  /* @ngInject */
+  constructor(
+    $q,
+    $translate,
+    OvhApiCloudProjectInstance,
+    PciProjectsProjectInstanceService,
+    Poller,
+  ) {
+    this.$q = $q;
+    this.$translate = $translate;
+    this.OvhApiCloudProjectInstance = OvhApiCloudProjectInstance;
+    this.PciProjectsProjectInstanceService = PciProjectsProjectInstanceService;
+    this.Poller = Poller;
+  }
+
+  $onInit() {
+    this.isLoading = false;
+  }
+
+  startPollerForShelveInstance() {
+    const endPointUrl = `/cloud/project/${this.projectId}/instance/${this.instance.id}`;
+    this.Poller.poll(endPointUrl, null, {
+      interval: 3000,
+      successRule(instance) {
+        return instance.status === 'SHELVED_OFFLOADED';
+      },
+      namespace: 'cloud.project.instance.shelve',
```

Comment:
Poller not destroyed when leaving the page ?

Reply:
no @marie-j , it destroy only if we change the universe.
The shelve and unshelve tasks, takes lot time.

Reply:
Yes but there's no need to poll if the user is not on the page right ? it takes resources for information the user doesn't need

Reply:
So, i let the poller run in the background until the shelve/unshelve task finished.
Once the back task is done, the instances/instance page will be refreshed.
The poller will be stopped once the shelve/unshelve process will finished.

Reply:
Thanks, i changed implementation as we discuss together and @frenautvh.
Fixed.